cpp-inference/inference_scaling_imagenet.cpp

```diff
@@ -86,30 +93,41 @@ void run_mnist(const std::string& model_name,
   int num_devices = get_num_devices();
   bool use_multigpu = (0 == device.compare("GPU")) && num_devices > 1;
   bool should_set = get_set_flag();
+  std::string model_key = "resnet_model";
+  std::string script_key = "resnet_script";
   if (should_set) {
     int batch_size = get_batch_size();
     int n_clients = get_client_count();
-    if (rank == 0) {
+    if (is_colo == false && rank == 0) {
       std::cout<<"Setting Resnet Model from scaling app" << std::endl << std::flush;
       std::cout<<"Setting with batch_size: " << std::to_string(batch_size) << std::endl << std::flush;
       std::cout<<"Setting on device: " << device << std::endl << std::flush;
       std::cout<<"Setting on " << std::to_string(num_devices) << " devices" <<std::endl << std::flush;
       std::string model_filename = "./resnet50." + device + ".pt";
 
       if (use_multigpu) {
-        std::string model_key = "resnet_model_0";
-        std::string script_key = "resnet_script_0";
         client.set_model_from_file_multigpu(model_key, model_filename, "TORCH", 0, num_devices, batch_size);
         client.set_script_from_file_multigpu(script_key, "./data_processing_script.txt", 0, num_devices);
       }
       else {
-        for (int i=0; i<num_devices; i++ ) {
-          std::string model_key = "resnet_model_" + std::to_string(i);
-          std::string script_key = "resnet_script_" + std::to_string(i);
           client.set_model_from_file(model_key, model_filename, "TORCH", device, batch_size);
           client.set_script_from_file(script_key, device, "./data_processing_script.txt");
+      }
+    }
+    if(is_colo && rank % n_clients == 0) {
+      std::cout<<"Setting Resnet Model from scaling app" << std::endl << std::flush;
+      std::cout<<"Setting with batch_size: " << std::to_string(batch_size) << std::endl << std::flush;
+      std::cout<<"Setting on device: " << device << std::endl << std::flush;
+      std::cout<<"Setting on " << std::to_string(num_devices) << " devices" <<std::endl << std::flush;
```

Comment:
one std::flush at the end of the last print statement is sufficient. There's no need to flush after each one

Reply:
done!